src/test/java/org/opensearch/sdk/TestExtensionsRunner.java

```diff
@@ -77,11 +78,14 @@ public class TestExtensionsRunner extends OpenSearchTestCase {
     private ExtensionsRunner extensionsRunner;
     private TransportService transportService;
 
+    private TransportService initialTransportService;
+
     @Override
     @BeforeEach
     public void setUp() throws Exception {
         super.setUp();
-        this.extensionsRunner = new ExtensionsRunner();
+        this.extensionsRunner = new ExtensionsRunnerForTest();
+        this.initialTransportService = extensionsRunner.extensionTransportService;
```

Comment:
The constructor of `ExtensionRunner` creates an instance of `TransportService` which listens a port in the background.

Tests further use `setExtensionTransportService` to replace the created instance with a mock to use in testing. The original instance becomes unreachable and our `tearDown` cannot stop it, which causes the rest of the tests to fail due to the port being already in use. Hence, this hack.

IMO, we should pass `TransportService` as a constructor parameter and not instantiate inside.

Reply:
At this point, let's "hack" as needed to get this merged.

I agree with you and have questioned how we've tracked the transport service from some of the initial PRs. I'm not sure putting it in the constructor is the best solution but we do need something better than what we have.

We do have a future requirement to gracefully handle disconnections/reconnections (hot plugin, etc.) and I think handling these will definitely need to be part of that functionality as well.